internal/provider/resource_subaccount_service_broker.go

```diff
@@ -0,0 +1,266 @@
+				Computed:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"name": schema.StringAttribute{
+				MarkdownDescription: "The name of the service broker.",
+				Required:            true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
+			},
+			"description": schema.StringAttribute{
+				MarkdownDescription: "The description of the service broker.",
+				Optional:            true,
+				Computed:            true,
+			},
+			"url": schema.StringAttribute{
+				MarkdownDescription: "The URL of the service broker.",
+				Required:            true,
+			},
+			"username": schema.StringAttribute{
+				MarkdownDescription: "The username for basic authentication against the service broker.",
+				Required:            true,
+			},
+			"password": schema.StringAttribute{
+				MarkdownDescription: "The password for basic authentication against the service broker.",
+				Required:            true,
+				Sensitive:           true,
+			},
+			"created_date": schema.StringAttribute{
+				MarkdownDescription: "The date and time when the resource was created in [RFC3339](https://www.ietf.org/rfc/rfc3339.txt) format.",
+				Computed:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"last_modified": schema.StringAttribute{
+				MarkdownDescription: "The date and time when the resource was last modified in [RFC3339](https://www.ietf.org/rfc/rfc3339.txt) format.",
+				Computed:            true,
+			},
+			"ready": schema.BoolAttribute{
+				MarkdownDescription: "Shows whether the service broker is ready.",
+				Computed:            true,
+			},
+		},
+	}
+}
+
+func (rs *subaccountServiceBrokerResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state subaccountServiceBrokerResourceType
+	diags := req.State.Get(ctx, &state)
+
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	cliRes, rawRes, err := rs.cli.Services.Broker.GetById(ctx, state.SubaccountId.ValueString(), state.Id.ValueString())
+	if err != nil {
+		handleReadErrors(ctx, rawRes, resp, err, "Resource Service Broker (Subaccount)")
+		return
+	}
+
+	newState := subaccountServiceBrokerValueFrom(ctx, cliRes)
+	newState.SubaccountId = state.SubaccountId
+	newState.Name = state.Name
+	newState.Username = state.Username
+	newState.Password = state.Password
+
+	diags = resp.State.Set(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+}
+
+func (rs *subaccountServiceBrokerResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan subaccountServiceBrokerResourceType
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	cliReq := btpcli.SubaccountServiceBrokerRegisterInput{
```

Comment:
Is there no polling needed i.e., the registration is fully available after the CREATE or UPDATE command was triggered?

Reply:
I'm not sure if it is required. But to be on the safe side we can add some logic to wait for the `ready` flag to become true.